core/src/main/scala/io/finch/route/RouterN.scala

```diff
@@ -41,9 +41,14 @@ trait RouterN[+A] { self =>
   def apply(route: Route): Option[(Route, A)]
 
   /**
+   * Attempts to match a route, but only returns any unmatched elements, not the value.
+   */
+  def exec(route: Route): Option[Route] = apply(route).map(_._1)
```

Comment:
That sounds reasonable to me.

I was thinking of creating an issue with a question about de-coupling router matching from returning the value (to help with cases like the issue I ran into in #302 that's fixed in #303)—i.e. if two routes match a request right now, both of their values will be evaluated (since this happens before the length of their leftovers are compared), which can be a little surprising. I was thinking this would happen (if at all) after 0.7.0, but it would fit in with the kind of changes you suggest.